packages/sample-app/sample_app/haystack_app.py

```diff
@@ -0,0 +1,28 @@
+import os
+from haystack import Pipeline
+from haystack.components.generators.chat import OpenAIChatGenerator
+from haystack.components.builders import DynamicChatPromptBuilder
+from haystack.dataclasses import ChatMessage
+from haystack.utils import Secret
+
+from traceloop.sdk import Traceloop
+
+Traceloop.init(app_name="haystack_example")
+
+def haystack_app():
+    
+    prompt_builder = DynamicChatPromptBuilder()
+    api_key=os.getenv("OPENAI_API_KEY")
+    llm = OpenAIChatGenerator(api_key=Secret.from_token(api_key), model="gpt-4")
+
+
+    pipe = Pipeline()
```

Comment:
So this is a really simple pipeline. I wonder if we should support more complex types of workflows like the ones Haystack publishes here - https://github.com/deepset-ai/haystack-cookbook/tree/main/notebooks

Reply:
Like this one - https://haystack.deepset.ai/tutorials/27_first_rag_pipeline

Reply:
Makes sense but do you think we should consider making this a separate PR?

Reply:
@mayankagarwals hmm. It just that this is a better example of what people do with Haystack. Otherwise I worry that when people will use this it may not work for them.

Reply:
Got it. Makes sense.